webpack-common.js

```diff
@@ -0,0 +1,173 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import config from 'config';
+import ExtractTextPlugin from 'extract-text-webpack-plugin';
+import webpack from 'webpack';
+
+import { getClientConfig } from 'core/utils';
+
+export function getRules({ babelQuery, bundleStylesWithJs = false } = {}) {
+  let styleRules;
+
+  if (bundleStylesWithJs) {
+    // In development, we bundle styles with the JS.
+    styleRules = [
+      {
+        test: /\.css$/,
+        use: [
+          { loader: 'style-loader' },
+          { loader: 'css-loader', options: { importLoaders: 2 } },
+          { loader: 'postcss-loader', options: { outputStyle: 'expanded' } },
+        ],
+      }, {
+        test: /\.scss$/,
+        use: [
+          { loader: 'style-loader' },
+          { loader: 'css-loader', options: { importLoaders: 2 } },
+          { loader: 'postcss-loader' },
+          { loader: 'sass-loader', options: { outputStyle: 'expanded' } },
+        ],
+      },
+    ];
+  } else {
+    // In production, we create a separate CSS bundle rather than
+    // include styles with the JS bundle. This lets the style bundle
+    // load in parallel.
+    styleRules = [
+      {
+        test: /\.css$/,
+        loader: ExtractTextPlugin.extract({
+          fallback: 'style-loader',
+          use: [
+            {
+              loader: 'css-loader',
+              options: { importLoaders: 2, sourceMap: true },
+            },
+            {
+              loader: 'postcss-loader',
+              options: {
+                outputStyle: 'expanded',
+                sourceMap: true,
+                sourceMapContents: true,
+              },
+            },
+          ],
+        }),
+      }, {
+        test: /\.scss$/,
+        loader: ExtractTextPlugin.extract({
+          fallback: 'style-loader',
+          use: [
+            {
+              loader: 'css-loader',
+              options: { importLoaders: 2, sourceMap: true },
+            },
+            { loader: 'postcss-loader' },
+            {
+              loader: 'sass-loader',
+              options: {
+                outputStyle: 'expanded',
+                sourceMap: true,
+                sourceMapContents: true,
+              },
+            },
+          ],
+        }),
+      },
+    ];
+  }
+
+  return [
+    {
+      test: /\.jsx?$/,
+      exclude: /node_modules/,
+      loader: 'babel-loader',
+      query: babelQuery,
+    },
+    ...styleRules,
+    {
+      test: /\.svg$/,
+      use: [{ loader: 'svg-url-loader', options: { limit: 10000 } }],
```

Comment:
I just read the docs on these loaders. I can add one here but why are we specifying mime types at all? The docs say that a mime type is inferred from file extension.